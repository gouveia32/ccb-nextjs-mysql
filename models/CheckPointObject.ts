export interface CheckPointInterface {
    id?: number | string;
    text: string;
    checked: boolean;
  }
  
  export type CheckPointType = CheckPointInterface;
  
  export const CheckPointObject: CheckPointType = {
           id: 0,
           text: "",
           checked: false,
         };
  
  export enum cCheckPoint {
    id = 'id',
    text = 'text',
    checked = 'checked',
  }