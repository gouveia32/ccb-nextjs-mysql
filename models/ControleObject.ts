export interface ControleInterface {
  id?: number | string;
  texto: string;
  marcado: boolean;
}

export type TipoControle = ControleInterface;

export const ControleObject: TipoControle = {
         id: 0,
         texto: "",
         marcado: false,
       };

export enum cControle {
  id = 'id',
  texto = 'texto',
  marcado = 'marcado',
}
