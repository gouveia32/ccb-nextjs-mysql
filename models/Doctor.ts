export interface DoctorModel {
  id: string;
  name: string;
  email: string;
  image?: string;
}

export type DoctorType = DoctorModel;

export const cDoctorModel = {
  id: "id",
  name: "name",
  email: "email",
  image: '',
};

export const DoctorObject: DoctorType = {
  id: "",
  name: "",
  email: "",
  image: '',
};

export function isDoctorType(object: any): object is DoctorType {
  return "name" in object;
}
