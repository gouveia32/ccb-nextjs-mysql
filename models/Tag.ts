export interface TagModel {
  id: string;
  name: string;
}

export type TagType = TagModel;

export const cTagModel = {
  id: "id",
  name: "name"
};

export const TagObject: TagType = {
  id: "",
  name: "",
};

export function isTagType(object: any): object is TagType {
  return "name" in object;
}
