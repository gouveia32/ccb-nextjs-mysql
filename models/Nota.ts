import { TipoPaciente } from "./Paciente";
import { TipoControle } from "./ControleObject";
import { Nota } from "@prisma/client";

export enum tipoNotaEnum {
  CHECK = "CHECK",
  TEXT = "TEXT",
}

export interface NotaModel {
  id?: string;
  nome: string;
  tipoNota: tipoNotaEnum;
  conteudo: string;
  cor: string;
  pino: boolean;
  criadoEm: string;
  pacientes: TipoPaciente[];
  controles?: TipoControle[];
}

export type TipoNota = NotaModel;

export const NotaObject: TipoNota = {
  id: "",
  nome: "",
  tipoNota: tipoNotaEnum.TEXT,
  conteudo: "",
  cor: "#fff",
  pino: false,
  criadoEm: new Date(Date.now()).toString(),
  pacientes: [],
  controles: [],
};

export const NotaDBObject: Nota = {
  id: "",
  nome: "",
  tipoNota: "TEXT",
  cor: "",
  conteudo: "",
  pino: false,
  criadoEm: new Date(Date.now()),
  medicoId: "",
};

export enum cNotaModel {
  id = "id",
  nome = "nome",
  tipoNota = "tipoNota",
  cor = "cor",
  conteudo = "conteudo",
  pino = "pino",
  criadoEm = "criadoEm",
  pacientes = "pacientes",
  controles = "controles",
}
