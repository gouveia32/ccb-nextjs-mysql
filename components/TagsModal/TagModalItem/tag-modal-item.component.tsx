import React, { useCallback, useEffect, useRef, useState } from "react";
import { TagType } from "../../../models/Tag";
import { Grid, IconButton, TextField } from "@material-ui/core";
import LabelOutlinedIcon from "@material-ui/icons/LabelOutlined";
import DeleteOutlineOutlinedIcon from "@material-ui/icons/DeleteOutlineOutlined";
import CheckOutlinedIcon from "@material-ui/icons/CheckOutlined";

export interface TagModalItemProps {
  tag: TagType;
  onDeleteTag: (tag: TagType) => void;
  onUpdateTag: (tag: TagType) => void;
}

const TagModalItem: React.FC<TagModalItemProps> = ({
  tag,
  onDeleteTag,
  onUpdateTag,
}: TagModalItemProps) => {
  const [edit, setEdit] = useState(false);

  const [updateTag, setUpdateTag] = useState(tag);

  const tagItemRef = useRef(null);

  const clickOutsideListener = useCallback(
    (e: MouseEvent) => {
      if (
        tagItemRef.current &&
        !(tagItemRef.current as any).contains(e.target)
      ) {
        setEdit(false);
      }
    },
    [tagItemRef]
  );

  useEffect(() => {
    document.addEventListener("click", clickOutsideListener);
    return () => {
      document.removeEventListener("click", clickOutsideListener);
    };
  }, [clickOutsideListener]);

  const handleOnChange = (value: string) => {
    setUpdateTag((prevState) => {
      const newTag = Object.assign({}, prevState);
      newTag.name = value;
      return newTag;
    });
  };

  const renderItems = (
    <Grid
      container={true}
      className="py-2 cursor-pointer"
      onClick={(e) => {
        setEdit(true);
        e.stopPropagation();
      }}
    >
      <Grid item={true}>
        <LabelOutlinedIcon />
      </Grid>
      <Grid item={true} style={{ flex: 1 }}>
        <h5 className="m-0 ms-2">{tag.name}</h5>
      </Grid>
      <Grid item={true}>
        <IconButton
          size={"small"}
          onClick={(event) => {
            onDeleteTag(tag);
            event.stopPropagation();
          }}
        >
          <DeleteOutlineOutlinedIcon />
        </IconButton>
      </Grid>
    </Grid>
  );

  const renderEditItems = (
    <Grid container={true} className="py-2">
      <Grid item={true}>
        <IconButton size={"small"} onClick={() => onDeleteTag(tag)}>
          <DeleteOutlineOutlinedIcon />
        </IconButton>
      </Grid>
      <Grid item={true} style={{ flex: 1 }}>
        <TextField
          value={updateTag.name}
          fullWidth={true}
          size={"small"}
          variant={"standard"}
          onChange={(event) => handleOnChange(event.target.value)}
          onKeyDown={(event) =>
            event.code === "Enter" && onUpdateTag(updateTag)
          }
        />
      </Grid>
      <Grid item={true}>
        <IconButton size={"small"} onClick={() => onUpdateTag(updateTag)}>
          <CheckOutlinedIcon />
        </IconButton>
      </Grid>
    </Grid>
  );

  return <div ref={tagItemRef}>{edit ? renderEditItems : renderItems}</div>;
};

export default React.memo(TagModalItem);
