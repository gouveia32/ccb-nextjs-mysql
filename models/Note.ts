import { TagType } from "./Tag";
import { CheckPointType } from "./CheckPointObject";
import { Note } from "@prisma/client";

export enum NoteTypeEnum {
  CHECK = "CHECK",
  TEXT = "TEXT",
}

export interface NoteModel {
  id?: string;
  name: string;
  noteType: NoteTypeEnum;
  content: string;
  color: string;
  pinned: boolean;
  createdAt: string;
  tags: TagType[];
  checkPoints?: CheckPointType[];
}

export type NoteType = NoteModel;

export const NoteObject: NoteType = {
  id: "",
  name: "",
  noteType: NoteTypeEnum.TEXT,
  content: "",
  color: "#fff",
  pinned: false,
  createdAt: new Date(Date.now()).toString(),
  tags: [],
  checkPoints: [],
};

export const NoteDBObject: Note = {
  id: "",
  name: "",
  noteType: "TEXT",
  color: "",
  content: "",
  pinned: false,
  createdAt: new Date(Date.now()),
  doctorId: "",
};

export enum cNoteModel {
  id = "id",
  name = "name",
  noteType = "noteType",
  color = "color",
  content = "content",
  pinned = "pinned",
  createdAt = "createdAt",
  tags = "tags",
  checkPoints = "checkPoints",
}
