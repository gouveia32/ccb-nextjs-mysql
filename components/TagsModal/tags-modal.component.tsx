import React, { useState } from "react";
import {
  Dialog,
  DialogContent,
  DialogTitle,
  Grid,
  IconButton,
  TextField,
} from "@material-ui/core";
import EditOutlinedIcon from "@material-ui/icons/EditOutlined";
import CheckOutlinedIcon from "@material-ui/icons/CheckOutlined";
import { cTagModel, TagType } from "../../models/Tag";
import { ChangeActionType } from "../../lib/helpers";
import NavigationItem from "../Navigation/NavItem/navitem.component";
import { Loading } from "../Loading/loading.component";
import TagModalItem from "./TagModalItem/tag-modal-item.component";

export interface TagsModalProps {
  newTag: TagType;
  tags: TagType[];
  tagsLoading: boolean;
  onChange: (value: ChangeActionType) => void;
  onAddTag: () => void;
  onUpdateTag: (tag: TagType) => void;
  onDeleteTag: (tag: TagType) => void;
}

const TagsModal: React.FC<TagsModalProps> = ({
  newTag,
  tags,
  tagsLoading,
  onChange,
  onAddTag,
  onUpdateTag,
  onDeleteTag,
}: TagsModalProps) => {
  const [open, setOpen] = useState<boolean>(false);

  const renderModal = (
    <Dialog open={open} onClose={() => setOpen(false)}>
      <DialogTitle>Edit tags</DialogTitle>
      <DialogContent>
        <Grid container={true} className="mb-2">
          <Grid item={true}>
            <TextField
              value={newTag.name}
              fullWidth={true}
              size={"small"}
              variant={"standard"}
              onChange={(event) =>
                onChange({ attr: cTagModel.name, value: event.target.value })
              }
              onKeyDown={(event) => event.code === "Enter" && onAddTag()}
            />
          </Grid>
          <Grid item={true}>
            <IconButton size={"small"} onClick={onAddTag}>
              <CheckOutlinedIcon />
            </IconButton>
          </Grid>
        </Grid>
        {tagsLoading ? (
          <Loading size={20} />
        ) : (
          tags &&
          tags.map((tag: TagType, k: number) => (
            <TagModalItem
              key={tag.id}
              tag={tag}
              onDeleteTag={onDeleteTag}
              onUpdateTag={onUpdateTag}
            />
          ))
        )}
      </DialogContent>
    </Dialog>
  );
  return (
    <>
      {renderModal}
      <NavigationItem
        name={"Altera tags"}
        onClick={() => setOpen((prevState) => !prevState)}
        icon={<EditOutlinedIcon />}
      />
    </>
  );
};

export default React.memo(TagsModal);
