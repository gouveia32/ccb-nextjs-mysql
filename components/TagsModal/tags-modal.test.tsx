import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { IconButton, TextField } from "@material-ui/core";
import TagsModal, { TagsModalProps } from "./tags-modal.component";
import { TagObject } from "../../models/Tag";
import TagModalItem from "./TagModalItem/tag-modal-item.component";

describe("TagsModal component", () => {
  let wrapper: ShallowWrapper;
  let mockOnChange: any;
  let mockOnAddTag: any;
  let mockOnUpdateTag: any;
  let mockOnDeleteTag: any;

  beforeEach(() => {
    mockOnChange = jest.fn();
    mockOnAddTag = jest.fn();
    mockOnUpdateTag = jest.fn();
    mockOnDeleteTag = jest.fn();

    const mockProps: TagsModalProps = {
      newTag: TagObject,
      tags: [],
      tagsLoading: false,
      onChange: mockOnChange,
      onAddTag: mockOnAddTag,
      onUpdateTag: mockOnUpdateTag,
      onDeleteTag: mockOnDeleteTag,
    };

    wrapper = shallow(<TagsModal {...mockProps} />);
  });

  it("should render TagsModal component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onChange on TextField component", () => {
    wrapper.find(TextField).simulate("change", { target: { value: "" } });
    expect(mockOnChange).toHaveBeenCalled();
  });

  it("should call onAddTag when IconButton component is clicked", () => {
    wrapper.find(IconButton).simulate("click");
    expect(mockOnAddTag).toHaveBeenCalled();
  });

  it("should render TagModalItem components when tags array is not empty and tags are not loading", () => {
    const mockProps: TagsModalProps = {
      newTag: TagObject,
      tags: [TagObject, TagObject],
      tagsLoading: false,
      onChange: mockOnChange,
      onAddTag: mockOnAddTag,
      onUpdateTag: mockOnUpdateTag,
      onDeleteTag: mockOnDeleteTag,
    };

    wrapper = shallow(<TagsModal {...mockProps} />);

    const tagsItems = wrapper.find(TagModalItem);

    expect(tagsItems).toHaveLength(2);
  });
});
