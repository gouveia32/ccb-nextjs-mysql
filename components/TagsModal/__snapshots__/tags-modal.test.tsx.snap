// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsModal component should render TagsModal component 1`] = `ShallowWrapper {}`;
